tests/test_northup.py

```diff
@@ -156,5 +156,50 @@ def test_northup(save_mock_dataset=False,save_derot_dataset=False,save_comp_figu
           plt.close(fig)
 
     return
+
+def test_wcs_and_offset(save_mock_dataset=False):
+   """
+   unit test of the create_wcs function and offset keyword 
+
+    Args:
+        save_mock_dataset (optional): if you want to save the original mock files at the input directory, turn True
+        
+   """
+   # read mock file
+   dirname = 'test_data/'
+   filename = 'JWST_CALFIELD2020.csv'
+
+   fieldpath = os.path.join(os.path.dirname(__file__),dirname,filename)
+   if not fieldpath:
+      raise FileNotFoundError(f"No filed data {filename} found")
+   
+   # running northup function
+   ang = 0
+   north_angle = 30
+   updated_datalist = []
+
+   # make a mock dataset
+   mock_dataset_ori = mocks.create_astrom_data(fieldpath, rotation=north_angle)
+   # run the boresight calibration to get an AstrometricCalibration file
+   astrom_cal = astrom.boresight_calibration(mock_dataset_ori, fieldpath, find_threshold=10)
+
+   mock_dataset =  mock_dataset_ori.copy()
+
+   # add an angle offset
+   mock_dataset[0].pri_hdr['ROLL']=(ang,'roll angle (deg)')
+
+   # create the wcs
+   updated_dataset = create_wcs(mock_dataset,astrom_cal,offset=(3.3, 1.0))
+   # inject fake sources for test
```

Comment:
there should be some assert statements to check that the function did what you expect (e.g., CRPIX is at a different location).